### PR TITLE
13272 update legend docs

### LINKED ISSUE #13272: update docs to show legend support for `nrows` and `ncols`
In the documentation a section about the usage of `nrows` and `ncols`  in a `Legend` is missing.
I want to add an example and write a short introduction.

### PR DESCRIPTION
**Pull request by mosc9575**
This pull request extends the documentation with information about two dimensional legends.

- [x] issues: fixes #13272 

The local build works fine.

### CODE REVIEW

**Comment by mattpap** [comment]
The first commit doesn't belong to this PR.

File: docs/bokeh/source/docs/user_guide/basic/annotations.rst [hunk 1]
```diff
@@ -170,6 +170,20 @@ either ``True`` or ``False``.
     and set the visibility of a ``legend_field`` item to ``False``, the entire
     group will be invisible.
 
+.. _ug_basic_annotations_legends_two_dimensions:
+
+Two dimensional legends
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To get a legend with more than one column or row, it is possible to set the
+``nrows`` or ``ncols`` property of the |Legend| with a position integer.
```
Comment (mosc9575) [inline docs/bokeh/source/docs/user_guide/basic/annotations.rst]:
```suggestion
``nrows`` or ``ncols`` property of the |Legend| with a positive integer.
```

File: examples/basic/annotations/legend_two_dimensions.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+'''Demonstrats how to modify a ``Legend`` to have multiple columns or rows by setting an
```
Comment (mosc9575) [inline examples/basic/annotations/legend_two_dimensions.py]:
```suggestion
'''Demonstrates how to modify a ``Legend`` to have multiple columns or rows by setting an
```

File: docs/bokeh/source/docs/user_guide/basic/annotations.rst (hunk 1 above)
Comment (bryevdv) [reply docs/bokeh/source/docs/user_guide/basic/annotations.rst]: typo: "positive"

File: docs/bokeh/source/docs/user_guide/styling/plots.rst [hunk 3]
```diff
@@ -708,6 +708,20 @@ The default orientation is ``"vertical"``.
 .. bokeh-plot:: __REPO__/examples/styling/plots/legend_orientation.py
     :source-position: above
 
+Two dimensional layout
+~~~~~~~~~~~~~~~~~~~~~~
+
+It is possible to activate a two dimensional layout for the legend by setting a positiv
```
Comment (mosc9575) [inline docs/bokeh/source/docs/user_guide/styling/plots.rst]:
```suggestion
It is possible to activate a two dimensional layout for the legend by setting a positive
```

File: docs/bokeh/source/docs/user_guide/basic/annotations.rst (hunk 1 above)
Comment (mosc9575) [reply docs/bokeh/source/docs/user_guide/basic/annotations.rst]: Thanks. I had this typo twice.

**Review by mattpap — APPROVED** [review]

**Review by ianthomas23 — APPROVED** [review]

**Pull request merged**